### PR TITLE
Write guide articles about generics

### LINKED ISSUE #1127: Document the generic type system
### Prerequisites

#1117 

### Content

- [ ] Add “Generic types and nodes” guide article to the “Concepts”
- [ ] Add “Making nodes generic” guide article to the “Making your own nodes”

### PR DESCRIPTION
**Pull request by nkrkv**
Closes #1127

### CODE REVIEW

File: docs/guide/creating-generics/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+To let XOD resolve generic nodes properly, you have to follow some rules while
+creating generic patch nodes and their specializations:
+
+1. If a patch contains generic _output_ terminal, a generic _input_ terminal of
+   the same generic type should present. In other words, if you use `output-t2`,
+   then `input-t2` should appear as well.
+2. There should be no more than a single specialization patch for a particular
+   generic patch. Otherwise, the choice will be ambiguous, and an error will be
+   shown. A xoder will be forced to resolve manually.
```
Comment (brusherru) [inline docs/guide/creating-generics/README.md]: "resolve" -> "choose specialization"?

File: docs/guide/generics/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+![Screencast](./screencast.gif)
+
+## Type deduction rules
+
+To be successfully resolved later a generic node should match a specific type to
+each of `t1`, `t2`, and `t3`. It is called _type deduction_.
+
+The deduction is done based on actual types of links and values bound to the
+generic pins. XOD will show an error if types can’t be deduced unambiguously,
+for example when one pin of type `t1` pretend to be number and another `t1` pin
```
Comment (brusherru) [inline docs/guide/generics/README.md]: Maybe add a hint (or a separate subsection), that if you want to use two different types in a generic node with casting one of the inputs, place cast nodes manually (`formatNumber`, `equals` and etc)?

File: docs/guide/generics/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+found:
+
+* `xod/core/if-else(number)`
+* `bob/stuff/if-else(number)`
+
+Now, if no specialization patches found XOD will show an error about unresolved
+abstract node. You have to create the specialization by yourself or install a
+library which includes proper specialization.
+
+If two or more specializations were found, that’s an error too. In that case,
```
Comment (evgenykochetkov) [inline docs/guide/generics/README.md]: I suggest adding a little explanation about *why* it's an error.

File: docs/guide/creating-generics/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+That forces XOD to search for a matching specialization patch when it processes
+the program. If a necessary specialization is not found, an error will show up,
+and the program will fail to transpile. Which is a good thing since it shows a
+xoder what’s missing so that he can provide an implementation for himself. See
+[generics resolution process](../generics/#resolution-process-overview) for
+details.
+
+## Specialization patches
+
+For a generic patch, you may provide several specialization patches. XOD search
```
Comment (evgenykochetkov) [inline docs/guide/creating-generics/README.md]: "XOD search**es** for ... every time"?

**Comment by nkrkv** [comment]
@brusherru @evgenykochetkov fixed

**Pull request merged**